### PR TITLE
search v2: pre-allocate dashboardQueryResult slice capacity for performance

### PR DESCRIPTION
**Pull request by mildwonkey**
I noticed that the search v2 dashboard index function has o(1) performance, with xorm spending thirty minutes (in a loadtesting instance!) appending to a slice of dashboardQueryResult. 

This PR pre-allocates the slice with a larger capacity using the `limit` parameter. The previous caller (`LoadDashboards`) already sets limit to 1 if searching for a single dashboard and otherwise uses search settings `DashboardLoadingBatchSize` to set the limit (thanks @timlevett!). 

This is the current performance on a load testing instance with a very large number (100K? @jalevin?) of dashboards: 
Screenshot 2024-08-05 at 9 58 22 AM

### CODE REVIEW

File: pkg/services/searchV2/index.go [hunk 1]
```diff
@@ -846,7 +846,14 @@ func (l sqlDashboardLoader) loadAllDashboards(ctx context.Context, limit int, or
...
 			))
 
-			rows := make([]*dashboardQueryResult, 0)
+			var rows []*dashboardQueryResult
+			if dashboardUID != "" { // only expecting a single result
+				// xorm will append if there is more than one
+				rows = make([]*dashboardQueryResult, 0, 1)
+			} else {
+				// start with an arbitrary but reasonable number of dashboards
+				rows = make([]*dashboardQueryResult, 0, 1000)
```
Comment (timlevett) [inline pkg/services/searchV2/index.go:855]: Instead of arbitrary, should we use the `limit` that is passed in?

Comment (mildwonkey) [reply pkg/services/searchV2/index.go:855]: Yes, that's a good point, I can use that when limit isn't set to 0, but we still need a good default value when we're returning all.

Comment (mildwonkey) [reply pkg/services/searchV2/index.go]: Oh, even better, now I see the default `limit` is tied to `DashboardLoadingBatchSize`, nice! Already configurable :)

**Review by jalevin — APPROVED** [review]

**Comment by jalevin** [comment]
@mildwonkey testing is done with 1m dashboards. Makes sense to me. Nice

**Pull request merged**